integrations/github/src/haystack_integrations/tools/github/file_editor_tool.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any, Dict, Optional
+
+from haystack import default_from_dict, default_to_dict
+from haystack.tools import ComponentTool
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+from haystack_integrations.components.connectors.github.file_editor import GitHubFileEditor
+from haystack_integrations.prompts.github.file_editor_tool import FILE_EDITOR_PROMPT, FILE_EDITOR_SCHEMA
+
+
+class GitHubFileEditorTool(ComponentTool):
+    """
+    A Haystack tool for editing files in GitHub repositories.
+    """
```

Comment:
Oh interesting I wasn't thinking to inherit from ComponentTool but do something like 
```python
GitHubFileEditorTool = ComponentTool(...)
```
and then people could import the pre-made `GitHubFileEditorTool` but I can see how this version would be more customizable.